lms/product/canvas/_plugin/course_copy.py

```diff
@@ -1,23 +1,18 @@
-from lms.services.exceptions import FileNotFoundInCourse
+from lms.product.plugin.course_copy import CourseCopyFilesHelper
 
 
 class CanvasCourseCopyPlugin:
     """Handle course copy in Canvas."""
 
-    def __init__(self, api, file_service):
+    file_type = "canvas_file"
+
+    def __init__(self, api, file_service, files_helper: CourseCopyFilesHelper):
         self._api = api
         self._file_service = file_service
+        self._files_helper = files_helper
 
     def is_file_in_course(self, course_id, file_id):
-        """Raise if the current user can't see file_id in course_id."""
-        for file in self._api.list_files(course_id):
-            # The Canvas API returns file IDs as ints but the file_id param
-            # that this method receives (from our proxy API) is a string.
-            # Convert ints to strings so that we can compare them.
-            if str(file["id"]) == file_id:
-                return
-
-        raise FileNotFoundInCourse("canvas_file_not_found_in_course", file_id)
+        return self._files_helper.is_file_in_course(course_id, file_id, self.file_type)
```

Comment:
This version checks against our DB instead of making a DB call.

Our DB could get out of sync with the API data but only having false positives (ie files that have been removed from the course since the assignment was configured).

These cases will still be handled by the `CanvasAPIPermissionError` catch in the service.

Reply:
I think in 100% of cases the logic here is the same. Every implementer of the `CourseCopyPlugin` interface has passes the call up to the files helper without modifying the call.

This makes me think it shouldn't be a customization point any more?

Also I think we are breaking with the general pattern established in other plugins, where the normal 80% case behavior is in the base class, and you only need to override the behavior you need to. In this case we have an abstract interface we don't inherit from and a helper to a plugin.

This means we have to test this boundary on every new instance, even though we don't implement any new behavior.

I think it might be worth trying a PR where:

 * `CourseCopyFilesHelper` and `CourseCopyGroupsHelper` are inlined into `CourseCopyPlugin` as a default behavior
 * `<Product>CourseCopyPlugin` instances become a child of `CourseCopyPlugin`
 * `<Product>CourseCopyPlugin` do not re-implement anything which is identical to the parent behavior
 
 I suspect this would result in quite a bit less code, and removes one extra layer of stuff.